### PR TITLE
stop propagation only after MvcEvent::EVENT_DISPATCH_ERROR

### PR DESCRIPTION
**Pull request by svycka**
because only one listener would be triggered
should fix https://github.com/ZF-Commons/zfc-rbac/issues/357

### CODE REVIEW

**Comment by prolic** [comment]
Can you add a test-case for this, please?

File: src/ZfcRbac/Guard/AbstractGuard.php [hunk 1]
```diff
@@ -68,18 +68,13 @@ public function onResult(MvcEvent $event)
...
-        if (method_exists($eventManager, 'triggerEvent')) {
-            // ZF3 EventManager
-            $event->setName(MvcEvent::EVENT_DISPATCH_ERROR);
-            $eventManager->triggerEvent($event);
-        } else {
-            // ZF2 EventManager
-            $eventManager->trigger(MvcEvent::EVENT_DISPATCH_ERROR, $event);
-        }
+        $event->setName(MvcEvent::EVENT_DISPATCH_ERROR);
+        $eventManager->triggerEvent($event);
```
Comment (svycka) [inline src/ZfcRbac/Guard/AbstractGuard.php]: just in case someone will come and complain about BC break. Yes this can be BC break because we depend on `EventManagerInterface` and if someone crazy enough implement it themselves and not use EventManager this is BC break but do we care? :)

**Comment by prolic** [comment]
Write a test that fails without this patch and succeeds with this patch.

File: tests/ZfcRbacTest/Guard/RoutePermissionsGuardTest.php [hunk 2]
```diff
@@ -410,7 +410,7 @@ public function testProperlyFillEventOnAuthorization()
 
     public function testProperlySetUnauthorizedAndTriggerEventOnUnauthorization()
     {
-        $eventManager = $this->getMock('Zend\EventManager\EventManagerInterface');
+        $eventManager = $this->getMock('Zend\EventManager\EventManager');
```
Comment (prolic) [inline tests/ZfcRbacTest/Guard/RoutePermissionsGuardTest.php]: Two questions:
1) Why is this needed?
2) How is this a BC break, when it's only in the tests?

**Comment by webimpress** [comment]
@svycka do we need still these conditions in tests:

```php
         if (method_exists($eventManager, 'triggerEvent')) {
             $eventManager->expects($this->once())
                          ->method('triggerEvent')
                          ->with($event);
         } else {
             $eventManager->expects($this->once())
                          ->method('trigger')
                          ->with(MvcEvent::EVENT_DISPATCH_ERROR);
         }
```
?

File: tests/ZfcRbacTest/Guard/RoutePermissionsGuardTest.php (hunk 2 above)
Comment (prolic) [inline tests/ZfcRbacTest/Guard/RoutePermissionsGuardTest.php]: Ah I see, because triggerEvent is not part of the 2.x series interface.

**Comment by prolic** [comment]
@webimpress correct, this could also be updated.

File: tests/ZfcRbacTest/Guard/RoutePermissionsGuardTest.php (hunk 2 above)
Comment (svycka) [inline tests/ZfcRbacTest/Guard/RoutePermissionsGuardTest.php]: because `EventManagerInterface` does not have `triggerEvent()` but `EventManager` has it

**Comment by svycka** [comment]
@prolic anything else?

**Comment by prolic** [comment]
Thx
